File: langs/python3/problems/rate_limiter/stub.py
class Simulation:
    def __init__(self):
        pass

    def allow(self, key, timestamp):
        """Spend one token. "true" if it fits, "false" if the window is full."""
        raise NotImplementedError

    def configure(self, key, limit, window):
        """Set that key's limit and window. "true" or "invalid_request"."""
        raise NotImplementedError

    def remaining(self, key, timestamp):
        """Unused tokens in this window as a string."""
        raise NotImplementedError

    def allow_weighted(self, key, cost, timestamp):
        """Spend cost tokens. "true", "false", or "invalid_request"."""
        raise NotImplementedError

File: langs/java/problems/rate_limiter/stub.java
public class Simulation {
    public Simulation() {}

    /**
     * Spend one token. {@code "true"} if it fits, {@code "false"} if the
     * window is full.
     */
    public String allow(String key, int timestamp) {
        return "";
    }

    /**
     * Set that key's limit and window. {@code "true"} or
     * {@code invalid_request}.
     */
    public String configure(String key, int limit, int window) {
        return "";
    }

    /**
     * Unused tokens in this window as a string.
     */
    public String remaining(String key, int timestamp) {
        return "";
    }

    /**
     * Spend cost tokens. {@code "true"}, {@code "false"}, or
     * {@code invalid_request}.
     */
    public String allowWeighted(String key, int cost, int timestamp) {
        return "";
    }
}
